File: enquiry-service/enquiry-service/src/main/java/com/booking/enquiryservice/entity/Enquiry.java
package com.booking.enquiryservice.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Data
public class Enquiry {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Relationships (Logical)
    private Long listingId;
    private Long vendorId;
    private Long coupleId;

    private LocalDate eventDate;
    private Integer guestCount;

    @Enumerated(EnumType.STRING)
    private EnquiryStatus status;

    private BigDecimal finalPrice;

    // --- Snapshots (Avoid N+1 calls) ---
    private String coupleNameSnapshot;
    private String couplePhoneSnapshot;
    private String listingTitleSnapshot;

    // --- Optimistic Locking ---
    @Version
    private Integer version;
}

File: frontend-app/Frontend/src/app/types/enquiry.ts
export interface Enquiry {
    id: number;
    listingId: number;
    vendorId: number;
    coupleId: number;
    eventDate: string; // ISO
    guestCount: number;
    status: 'PENDING' | 'ACCEPTED' | 'DECLINED' | 'BOOKED';
    finalPrice?: number;
    message?: string;
    // Snapshots
    coupleNameSnapshot?: string;
    coupleEmailSnapshot?: string;
    couplePhoneSnapshot?: string;
    listingTitleSnapshot?: string;
}

export interface EnquiryRequest {
    listingId: number;
    vendorId: number;
    coupleId: number;
    eventDate: string;
    guestCount: number;
    message?: string;
    // Snapshots to ensure backend saves them
    coupleNameSnapshot?: string;
    coupleEmailSnapshot?: string;
    couplePhoneSnapshot?: string;
    listingTitleSnapshot?: string;
}
